### ISSUE TITLE
[Bug]: query offset larger than number of entities cause proxy crash

### ISSUE THREAD

**Issue by xiaoFine** [milvus-io/milvus#25299]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:v2.2.2
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka): pulsar
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.2.2
- OS(Ubuntu or CentOS): macOS
- CPU/Memory: 
- GPU: 
- Others: deployed by helm
```

### Current Behavior

proxy crashed when the `offset` parameter is set more than the amount of query results
![mWOxvxvoea](https://github.com/milvus-io/milvus/assets/12167470/5e11469a-1de7-4878-b15b-d7097cc4158a)

crash log
`kubectl logs  milvus-proxy-6bcd477855-8x2rj -n milvus -c proxy --previous`
```bash
...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2bd1fb1]

goroutine 446090 [running]:
github.com/milvus-io/milvus/internal/proxy.(*queryTask).PostExecute(0xc001cc0fa0, {0x40fbf98, 0xc0021cb9b0})
	/go/src/github.com/milvus-io/milvus/internal/proxy/task_query.go:388 +0xff1
github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).processTask(0x2526540?, {0x4113580, 0xc001cc0fa0}, {0x4111030, 0xc000523d40})
	/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:468 +0xbe8
created by github.com/milvus-io/milvus/internal/proxy.(*taskScheduler).queryLoop
	/go/src/github.com/milvus-io/milvus/internal/proxy/task_scheduler.go:519 +0x1e5
...
```

collection schmea:

```
{
  auto_id: True
  description: 
  fields: [{
    name: id
    description: 
    type: 5
    is_primary: True
    auto_id: True
  }, {
    name: embedding
    description: 
    type: 101
    params: {'dim': 1024}
  }, {
    name: url
    description: 
    type: 21
    params: {'max_length': 1000}
  }, {
    name: raw_id
    description: 
    type: 5
  }]
```

**Comment by yanliang567** [comment]
/assign @NicoYuan1986 
could you please help to reproduce it on latest 2.2.x release

/unassign

**Comment by NicoYuan1986** [comment]
![image](https://github.com/milvus-io/milvus/assets/109071306/bdb8e265-a351-4680-9a9a-7385c8abd174)

not reproduced on v2.2.11
Maybe you can use the latest version. @xiaoFine

**Comment by xiaoFine** [comment]
> ![image](https://user-images.githubusercontent.com/109071306/250741294-bdb8e265-a351-4680-9a9a-7385c8abd174.png)
> 
> not reproduced on v2.2.11 Maybe you can use the latest version. @xiaoFine

could you please use expr  like `'url !=""'` to show the number of records

**Comment by xiaoFine** [comment]
the milvus has been deployed to product env and I wonder if it's ok to upgrade 

or any suggestion to invest this bug on 2.2.2 ?

**Comment by xiaofan-luan** [comment]
seem as https://github.com/milvus-io/milvus/issues/21223

**Comment by xiaoFine** [comment]
> seem as #21223

yes,  except the deployment mode

it seems the `offset` problem has been fix in 2.2.0
